gpt4all-bindings/python/gpt4all/gpt4all.py

```diff
@@ -33,12 +33,15 @@ def __init__(self, model_name: str, model_path: str = None, model_type: str = No
             model_type: Model architecture. This argument currently does not have any functionality and is just used as
                 descriptive identifier for user. Default is None.
             allow_download: Allow API to download models from gpt4all.io. Default is True. 
+            n_threads: number of CPU threads used by GPT4All
         """
         self.model_type = model_type
         self.model = pyllmodel.LLModel()
         # Retrieve model and download if allowed
         model_dest = self.retrieve_model(model_name, model_path=model_path, allow_download=allow_download)
         self.model.load_model(model_dest)
+        # Set n_threads
+        self.model.set_thread_count(n_threads)
```

Comment:
internally the library default may be _less_ than 4 if there are fewer than 4 cores to detected _or_ if one of the GPU implementations is being used, it would be better to make n_threads `Optional` / default None and only call `set_thread_count` / override the library's default if the user actually passes something intentionally

Reply:
changed that